operator/controllers/operator/eventexporter_controller.go

```diff
@@ -0,0 +1,210 @@
+	l "github.com/sirupsen/logrus"
+	apps "k8s.io/api/apps/v1"
+	core "k8s.io/api/core/v1"
+	apiequal "k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	runtimelog "sigs.k8s.io/controller-runtime/pkg/log"
+	"text/template"
+
+	operatorv1alpha1 "github.com/apache/skywalking-swck/operator/apis/operator/v1alpha1"
+	"github.com/apache/skywalking-swck/operator/pkg/kubernetes"
+)
+
+// EventExporterReconciler reconciles a EventExporter object
+type EventExporterReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	FileRepo kubernetes.Repo
+	Recorder record.EventRecorder
+}
+
+// +kubebuilder:rbac:groups=operator.skywalking.apache.org,resources=eventexporters,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=operator.skywalking.apache.org,resources=eventexporters/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=operator.skywalking.apache.org,resources=eventexporters/finalizers,verbs=update
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list
+// +kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+
+func (r *EventExporterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := runtimelog.FromContext(ctx)
+	log.Info("=====================eventexporter reconcile started================================")
+
+	eventExporter := operatorv1alpha1.EventExporter{}
+	if err := r.Client.Get(ctx, req.NamespacedName, &eventExporter); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	if _, err := r.overlayData(ctx, log, &eventExporter); err != nil {
+		l.Error(err, "failed to overlay eventexporter's configMap")
+		return ctrl.Result{}, err
+	}
+
+	ff, err := r.FileRepo.GetFilesRecursive("templates")
+	if err != nil {
+		log.Error(err, "failed to load resource templates")
+		return ctrl.Result{}, err
+	}
+
+	app := kubernetes.Application{
+		Client:   r.Client,
+		FileRepo: r.FileRepo,
+		CR:       &eventExporter,
+		GVK:      operatorv1alpha1.GroupVersion.WithKind("EventExporter"),
+		Recorder: r.Recorder,
+		TmplFunc: template.FuncMap{
+			"md5Data": func() string { return MD5Hash(eventExporter.Spec.Data) },
+		},
+	}
+
+	if err := app.ApplyAll(ctx, ff, log); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.checkState(ctx, log, &eventExporter); err != nil {
+		l.Error(err, "failed to check sub resources state")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{RequeueAfter: schedDuration}, nil
+
+}
+
+func (r *EventExporterReconciler) overlayData(ctx context.Context, log logr.Logger, eventExporter *operatorv1alpha1.EventExporter) (bool, error) {
```

Comment:
Let's give the returned variables names, like `(changed bool, err error)` for better readability

Reply:
Thx, updated!